checks.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# As described in `CONTRIBUTING.md`, this script checks Optuna's source codes for formatting,
+# coding style and type hints. The `-u` option enables automatic formatting by `black` and `isort`.
+
+res_pip_list=$(pip freeze)
+missing_dependencies=()
+if [ ! "$(echo $res_pip_list | grep black)" ] ; then
+  missing_dependencies+=(black)
+fi
+if [ ! "$(echo $res_pip_list | grep flake8)" ] ; then
+  missing_dependencies+=(hacking)
+fi
+if [ ! "$(echo $res_pip_list | grep isort)" ] ; then
+  missing_dependencies+=(isort)
+fi
+if [ ! "$(echo $res_pip_list | grep mypy)" ] ; then
+  missing_dependencies+=(mypy)
+fi
+if [ ! ${#missing_dependencies[@]} -eq 0 ]; then
+  echo "The following dependencies are missing:" "${missing_dependencies[@]}"
+  read -p "Would you like to install the missing dependencies? (y/N): " yn
+  case "$yn" in [yY]*) ;; *) echo "abort." ; exit ;; esac
+  pip install "${missing_dependencies[@]}"
+This conversation was marked as resolved by hvy
+fi
+
+update=0
+while getopts "u" OPT
+do
+  case $OPT in
+    u) update=1
+       ;;
+    *) ;;
+  esac
+done
+
+target="examples optuna tests"
+mypy_target="optuna tests"
+res_all=0
+
+res_black=$(black $target --check --diff 2>&1)
+if [ $? -eq 1 ] ; then
+  if [ $update -eq 1 ] ; then
+    echo "black failed. The code will be formatted by black."
+    black .
+  else
+    echo "$res_black"
+    echo "black failed."
+    res_all=1
+  fi
+else
+  echo "black succeeded."
+fi
+
+res_flake8=$(flake8 $target)
+if [ $? -eq 1 ] ; then
+  echo "$res_flake8"
+  echo "flake8 failed."
+  res_all=1
+fi
+echo "flake8 succeeded."
```

Comment:
When `flake8` failed, I got an inconsistent message as follows:
```console
2     F401 'numpy as np' imported but unused
flake8 failed.
flake8 succeeded.
```

I guess we need to add `else` as follows:

```suggestion
else
  echo "flake8 succeeded."
fi
```